include/crocoddyl_msgs/whole_body_state_publisher.h

```diff
@@ -137,6 +137,46 @@ class WholeBodyStateRosPublisher {
     }
   }
 
+  /**
+   * @brief Publish a whole-body state ROS message
+   *
+   * @param t[in]    Time in secs
+   * @param q[in]    Configuration vector (dimension: model.nq)
+   * @param v[in]    Generalized velocity (dimension: model.nv)
+   * @param a[in]    Generalized acceleration (dimension: model.nv)
+   * @param tau[in]  Joint effort (dimension: model.nv)
+   * @param p[in]    Contact position
+   * @param pd[in]   Contact velocity
+   * @param f[in]    Contact force, type and status
+   * @param s[in]    Contact surface and friction coefficient
+   */
+  void
+  publish(const double t, const Eigen::Ref<const Eigen::VectorXd> &q,
+          const Eigen::Ref<const Eigen::VectorXd> &v,
+          const Eigen::Ref<const Eigen::VectorXd> &a,
+          const Eigen::Ref<const Eigen::VectorXd> &tau,
+          const std::map<std::string, pinocchio::SE3> &p,
+          const std::map<std::string, pinocchio::Motion> &pd,
+          const std::map<std::string, std::tuple<pinocchio::Force, ContactType,
+                                                 ContactStatus>> &f,
+          const std::map<std::string, std::pair<Eigen::Vector3d, double>> &s) {
+    if (pub_.trylock()) {
+      pub_.msg_.header.frame_id = odom_frame_;
+      crocoddyl_msgs::toMsg(model_, data_, pub_.msg_, t, q, v, a, tau, p, pd,
+                            f, s);
+      pub_.unlockAndPublish();
+    }
+  }
+  
+  /**
+   * @brief Returns a pointer to the pinocchio model
+   *
+   * @return pointer to the pinocchio model
+   */
+  pinocchio::Model* get_pinocchio_model(){
+    return &model_;
+  }
```

Comment:
Returning points is dangerous as the users might make mistakes. We should return a reference, instead.
```suggestion
  /**
   * @brief Return the Pinocchio model (i.e., model of the rigid body system)
   */
  pinocchio::Model& get_model() {
    return model_;
  }
```